### ISSUE TITLE
Calling a contract's function returns 0 in Nethereum but a valid value in JS

### ISSUE THREAD

**Issue by thomasd3** [Nethereum/Nethereum#668]
The same call to a contract in JS and Nethereum yields different results:

in F#:

let pancakeSwapAddress = "0x73feaa1eE314F8c655E354234017bE2193C9E24E"   
let contract = web3.Eth.GetContract(abi, pancakeSwapAddress)
let cakePerBlockFunc = contract.GetFunction("cakePerBlock")
let cakePerBlock = cakePerBlockFunc.CallAsync<int64>() |> Async.AwaitTask |> Async.RunSynchronously

will return 0

and, in JS:

const PANCAKE_CHEF_ADDR = "0x73feaa1eE314F8c655E354234017bE2193C9E24E";
const PANCAKE_CHEF = new ethers.Contract(PANCAKE_CHEF_ADDR, PANCAKE_CHEF_ABI, App.provider);
let cakerPerBlock = await PANCAKE_CHEF.cakePerBlock()

this will return 40000000000000000000

the ABI is long so I put it there: https://pastebin.com/0CTitM25

I tried different return types (int, int64, BigInteger, string, byte[], etc) and it is still the same problem.

**Comment by juanfranblanco** [comment]
Have you tried to code generate it as I mentioned on the chat, to validate your code is correct? Are you able to do other calls to other contracts in Ethereum mainnet or other test chain?

**Comment by thomasd3** [comment]
I've used the code generation and it provided me the definitions and services. Then I called it with:

let p = PancakeService (web3, pancakeSwapAddress)
let x = p.CakePerBlockQueryAsync(CakePerBlockFunction()) |> Async.AwaitTask |> Async.RunSynchronously

(like I posted on the Discord chat, but having a copy here may help people with a similar problem)

it still returns 0. What I don't get is that it seems identical to the JS code, I don't see any difference at all. Unfortunately Nethereum doesn't build on my mac, otherwise I'd trace the call to see if there is anything returned from the chain, before it's processed by Nethereum.
